src/CSI.IBTA.Administrator/Controllers/HomeController.cs

```diff
@@ -38,21 +47,75 @@ public HomeController(IUserServiceClient userServiceClient)
             if (res.Result != null)
             {
                 var employers = res.Result;
-                if (!String.IsNullOrEmpty(nameFilter))
+
+                if (!string.IsNullOrEmpty(nameFilter))
                 {
                     employers = employers.Where(s => s.Name.Contains(nameFilter));
                 }
-                if (!String.IsNullOrEmpty(codeFilter))
+                if (!string.IsNullOrEmpty(codeFilter))
                 {
                     employers = employers.Where(s => s.Code.Equals(codeFilter));
                 }
+
                 ViewData["Page"] = "Home";
-                return (View(new PaginatedList<EmployerDto>(employers ?? new List<EmployerDto>().AsQueryable(), pageNumber ?? 1, pageSize ?? 8)));
+                var employerList = employers ?? new List<EmployerDto>().AsQueryable();
+                var paginatedList = new PaginatedList<EmployerDto>(employerList, pageNumber ?? 1, pageSize ?? 8);
+                return PartialView("_Employer", paginatedList);
             }
+
             if (res.Error.StatusCode == HttpStatusCode.Unauthorized)
+            {
                 return RedirectToAction("Index", "Auth");
+            }
 
-            return View("Index");
+            return PartialView("_Employer");
+        }
+
+        [HttpGet("Claims")]
+        public async Task<IActionResult> GetClaims(
+            string? numberFilter,
+            string? employerFilter,
+            string? currentNumberFilter,
+            string? currentEmployerFilter,
+            int? pageNumber,
+            int? pageSize)
+        {
+            if (numberFilter != null || employerFilter != null)
+            {
+                pageNumber = 1;
+            }
+
+            numberFilter = numberFilter ?? currentNumberFilter;
+            employerFilter = employerFilter ?? currentEmployerFilter;
+            ViewData["CurrentNumberFilter"] = numberFilter;
+            ViewData["CurrentEmployerFilter"] = employerFilter;
+
+            var res = await _claimsClient.GetClaims();
```

Comment:
The code pulls all the claims into a memory and after that it starts to filter them. 
The claims should be filtered out on the server side, and GetClaims should accept filters. 
Preferably, it does paging too, but it is acceptable to have paging on controller side for now.

Reply:
I was using old version of view Claims functionality, this issue has been fixed by merging with IBTA-29 branch latest commit